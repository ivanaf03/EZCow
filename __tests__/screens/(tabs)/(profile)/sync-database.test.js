import React from "react";

import { render, fireEvent } from "@testing-library/react-native";
import { Alert } from "react-native";

import SyncDatabase from "../../../../app/(tabs)/(profile)/sync-database";

jest.mock("@fortawesome/react-native-fontawesome", () => ({
  FontAwesomeIcon: () => null,
}));

jest.mock("../../../../app/model/users", () => ({
  getUserByEmail: jest.fn(),
}));

jest.mock("../../../../app/model/bd", () => ({
  closeDatabase: jest.fn(),
  syncDatabase: jest.fn(),
  getDatabase: jest.fn(),
}));

jest.spyOn(Alert, "alert");

describe("SyncDatabase", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("should render correctly", async () => {
    const tree = render(<SyncDatabase />);
    expect(tree).toMatchSnapshot();
  });

  it("should show an error if device name is empty", async () => {
    const { getByPlaceholderText, getByTestId } = render(<SyncDatabase />);
    const deviceNameInput = getByPlaceholderText("Nombre del dispositivo");
    fireEvent.changeText(deviceNameInput, "");
    const passwordInput = getByPlaceholderText("Contrase単a");
    fireEvent.changeText(passwordInput, "12345678");
    const syncDatabaseButton = getByTestId("sync-database-button");
    fireEvent.press(syncDatabaseButton);
    expect(Alert.alert).toHaveBeenCalledWith(
      "Error",
      "Por favor, rellena todos los campos."
    );
  });

  it("should show an error if password is empty", async () => {
    const { getByPlaceholderText, getByTestId } = render(<SyncDatabase />);
    const deviceNameInput = getByPlaceholderText("Nombre del dispositivo");
    fireEvent.changeText(deviceNameInput, "TestDevice");
    const passwordInput = getByPlaceholderText("Contrase単a");
    fireEvent.changeText(passwordInput, "");
    const syncDatabaseButton = getByTestId("sync-database-button");
    fireEvent.press(syncDatabaseButton);
    expect(Alert.alert).toHaveBeenCalledWith(
      "Error",
      "Por favor, rellena todos los campos."
    );
  });

  it("should show an error if password is less than 8 characters", async () => {
    const { getByPlaceholderText, getByTestId } = render(<SyncDatabase />);
    const deviceNameInput = getByPlaceholderText("Nombre del dispositivo");
    fireEvent.changeText(deviceNameInput, "TestDevice");
    const passwordInput = getByPlaceholderText("Contrase単a");
    fireEvent.changeText(passwordInput, "1234567");
    const syncDatabaseButton = getByTestId("sync-database-button");
    fireEvent.press(syncDatabaseButton);
    expect(Alert.alert).toHaveBeenCalledWith(
      "Error",
      "La contrase単a debe tener al menos 8 caracteres."
    );
  });
});
