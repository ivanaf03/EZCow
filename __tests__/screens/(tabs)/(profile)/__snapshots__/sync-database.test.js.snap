// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SyncDatabase should render correctly 1`] = `
<RCTScrollView
  style={
    [
      {
        "flexBasis": "0%",
        "flexGrow": 1,
        "flexShrink": 1,
      },
      {
        "backgroundColor": "#1c1c1c",
      },
    ]
  }
>
  <View>
    <Text
      style={
        [
          {
            "marginTop": 16,
          },
          {
            "marginLeft": 16,
            "marginRight": 16,
          },
          {
            "color": "#ffffff",
          },
          {
            "fontSize": 36,
            "lineHeight": 40,
          },
          {
            "fontFamily": "Nunito-Bold",
          },
          {
            "borderBottomWidth": 2,
          },
          {
            "borderBottomColor": "#ffffff",
            "borderLeftColor": "#ffffff",
            "borderRightColor": "#ffffff",
            "borderTopColor": "#ffffff",
          },
        ]
      }
    >
      Sincronizar base de datos
    </Text>
    <View
      style={
        [
          {
            "display": "flex",
          },
          {
            "alignItems": "center",
          },
          {
            "marginTop": 32,
          },
        ]
      }
    >
      <View
        style={
          [
            {
              "width": "90%",
            },
          ]
        }
      >
        <View
          style={
            [
              {
                "display": "flex",
              },
              {
                "alignItems": "center",
              },
              {
                "justifyContent": "center",
              },
              {
                "paddingBottom": 16,
                "paddingLeft": 16,
                "paddingRight": 16,
                "paddingTop": 16,
              },
              {
                "backgroundColor": "#272727",
              },
              {
                "borderLeftWidth": 2,
                "borderRightWidth": 2,
              },
              {
                "borderBottomColor": "#6bc0eb",
                "borderLeftColor": "#6bc0eb",
                "borderRightColor": "#6bc0eb",
                "borderTopColor": "#6bc0eb",
              },
              {
                "borderBottomLeftRadius": 8,
                "borderBottomRightRadius": 8,
                "borderTopLeftRadius": 8,
                "borderTopRightRadius": 8,
              },
            ]
          }
        >
          <Text
            style={
              [
                {
                  "color": "#ffffff",
                },
                {
                  "fontSize": 24,
                  "lineHeight": 32,
                },
                {
                  "fontFamily": "Nunito-Medium",
                },
              ]
            }
          >
            ◆ Cómo funciona la sincronización?
          </Text>
          <Text
            style={
              [
                {
                  "marginTop": 8,
                },
                {
                  "color": "#ffffff",
                },
                {
                  "fontFamily": "Nunito-Italic",
                },
              ]
            }
          >
            Si tienes una copia de seguridad de tu base de datos subida al servidor, puedes introducir el nombre de dispositivo que usaste para subirla y la contraseña asociada para actualizarla. También puedes utilizar un nuevo nombre de dispositivo y contraseña para guardar otra copia diferente. En cualquier caso, recuerda tus datos para poder recuperar tus copias de seguridad en un futuro.
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        [
          {
            "marginTop": 32,
          },
          {
            "paddingLeft": 16,
            "paddingRight": 16,
          },
        ]
      }
    >
      <View
        style={
          [
            {
              "display": "flex",
            },
            {
              "paddingBottom": 16,
              "paddingLeft": 16,
              "paddingRight": 16,
              "paddingTop": 16,
            },
            {
              "backgroundColor": "#272727",
            },
            {
              "borderBottomLeftRadius": 8,
              "borderBottomRightRadius": 8,
              "borderTopLeftRadius": 8,
              "borderTopRightRadius": 8,
            },
          ]
        }
      >
        <View>
          <View
            style={
              [
                {
                  "marginTop": -4,
                },
              ]
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#ffffff",
                  },
                  {
                    "fontSize": 14,
                    "lineHeight": 20,
                  },
                  {
                    "fontFamily": "Nunito-Medium",
                  },
                  [
                    {
                      "marginTop": 4,
                    },
                  ],
                ]
              }
            >
              Nombre del dispositivo
              :
            </Text>
            <TextInput
              onChangeText={[Function]}
              placeholder="Nombre del dispositivo"
              style={
                [
                  {
                    "paddingBottom": 8,
                    "paddingLeft": 8,
                    "paddingRight": 8,
                    "paddingTop": 8,
                  },
                  {
                    "backgroundColor": "#ffffff",
                  },
                  {
                    "borderBottomWidth": 2,
                    "borderLeftWidth": 2,
                    "borderRightWidth": 2,
                    "borderTopWidth": 2,
                  },
                  {
                    "borderBottomColor": "#2254ba",
                    "borderLeftColor": "#2254ba",
                    "borderRightColor": "#2254ba",
                    "borderTopColor": "#2254ba",
                  },
                  {
                    "borderBottomLeftRadius": 16,
                    "borderBottomRightRadius": 16,
                    "borderTopLeftRadius": 16,
                    "borderTopRightRadius": 16,
                  },
                  {
                    "fontFamily": "Nunito-Medium",
                  },
                  [
                    {
                      "marginTop": 4,
                    },
                  ],
                ]
              }
              value=""
            />
          </View>
        </View>
        <View
          style={
            [
              {
                "marginTop": 24,
              },
            ]
          }
        >
          <View
            style={
              [
                {
                  "marginTop": -4,
                },
              ]
            }
          >
            <Text
              style={
                [
                  {
                    "color": "#ffffff",
                  },
                  {
                    "fontSize": 14,
                    "lineHeight": 20,
                  },
                  {
                    "fontFamily": "Nunito-Light",
                  },
                  [
                    {
                      "marginTop": 4,
                    },
                  ],
                ]
              }
            >
              Contraseña
              :
            </Text>
            <View
              style={
                [
                  {
                    "flexDirection": "row",
                  },
                  {
                    "alignItems": "center",
                  },
                  [
                    {
                      "marginTop": 4,
                    },
                  ],
                ]
              }
            >
              <TextInput
                onChangeText={[Function]}
                placeholder="Contraseña"
                secureTextEntry={true}
                style={
                  [
                    {
                      "paddingBottom": 8,
                      "paddingLeft": 8,
                      "paddingRight": 8,
                      "paddingTop": 8,
                    },
                    {
                      "marginRight": 8,
                    },
                    {
                      "backgroundColor": "#ffffff",
                    },
                    {
                      "borderBottomWidth": 2,
                      "borderLeftWidth": 2,
                      "borderRightWidth": 2,
                      "borderTopWidth": 2,
                    },
                    {
                      "borderBottomColor": "#2254ba",
                      "borderLeftColor": "#2254ba",
                      "borderRightColor": "#2254ba",
                      "borderTopColor": "#2254ba",
                    },
                    {
                      "borderBottomLeftRadius": 16,
                      "borderBottomRightRadius": 16,
                      "borderTopLeftRadius": 16,
                      "borderTopRightRadius": 16,
                    },
                    {
                      "flexBasis": "0%",
                      "flexGrow": 1,
                      "flexShrink": 1,
                    },
                    {
                      "fontFamily": "Nunito-Medium",
                    },
                  ]
                }
                value=""
              />
              <View
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                testID="eye-icon"
              />
            </View>
          </View>
        </View>
        <View
          style={
            [
              {
                "marginTop": 24,
              },
            ]
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "paddingBottom": 16,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "paddingTop": 16,
                },
                {
                  "backgroundColor": "#272727",
                },
                {
                  "borderLeftWidth": 8,
                  "borderRightWidth": 8,
                },
                {
                  "borderBottomWidth": 2,
                  "borderTopWidth": 2,
                },
                {
                  "borderBottomColor": "#2254ba",
                  "borderLeftColor": "#2254ba",
                  "borderRightColor": "#2254ba",
                  "borderTopColor": "#2254ba",
                },
                {
                  "borderBottomLeftRadius": 24,
                  "borderBottomRightRadius": 24,
                  "borderTopLeftRadius": 24,
                  "borderTopRightRadius": 24,
                },
              ]
            }
            testID="sync-database-button"
          >
            <Text
              style={
                [
                  {
                    "fontFamily": "Nunito-SemiBold",
                  },
                  {
                    "color": "#ffffff",
                  },
                  {
                    "fontSize": 20,
                    "lineHeight": 28,
                  },
                  {
                    "textAlign": "center",
                  },
                ]
              }
            >
              Guardar copia de seguridad
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
